File: client/src/lib/invoke/getUserStats.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Config, UserStats } from "@bindings";

export function getUserStats(login: string, config: Config): ResultAsync<UserStats, string> {
    return ResultAsync.fromPromise(
        invoke("get_user_stats", { login, config }),
        (error) => `Failed to get user stats: ${error}`
    );
}

File: client/src-tauri/src/commands/get_user_stats.rs
use crate::core::{
    config::Config,
    nexium_api::{get_user_stats as get_user_stats_api, UserStats},
};

#[tauri::command]
pub async fn get_user_stats(
    login: String,
    config: Config,
) -> Result<UserStats, String> {
    tauri::async_runtime::spawn_blocking(move || {
        get_user_stats_api(login, config)
    })
    .await
    .map_err(|err| format!("Failed to execute get_user_stats: {}", err))?
}
